ape_safe/accounts.py

```diff
@@ -822,33 +840,48 @@ def skip_signer(signer: AccountAPI):
         return None
 
     def add_signatures(
-        self, safe_tx: SafeTx, confirmations: Optional[list[SafeTxConfirmation]] = None
+        self,
+        safe_tx: Union[SafeTx, SafeTxID],
+        confirmations: Optional[list[SafeTxConfirmation]] = None,
     ) -> dict[AddressType, MessageSignature]:
-        confirmations = confirmations or []
-        if not self.local_signers:
-            raise ApeSafeError("Cannot sign without local signers.")
-
-        amount_needed = self.confirmations_required - len(confirmations)
-        signers = [
-            acc for acc in self.local_signers if acc.address not in [c.owner for c in confirmations]
-        ][:amount_needed]
-
-        safe_tx_hash = _get_safe_tx_id(safe_tx, confirmations)
-        signatures = get_signatures(safe_tx, signers)
-        if signatures:
-            self.client.post_signatures(safe_tx_hash, signatures)
+        if not isinstance(safe_tx, (SafeTxV1, SafeTxV2)):
+            safe_tx_id = safe_tx
+            safe_tx = self.client.get_safe_tx(safe_tx_id).as_safe_tx(
+                version=str(self.version), chain_id=self.chain_manager.chain_id
+            )
 
-        return signatures
+        else:
+            safe_tx_id = get_safe_tx_hash(safe_tx)
+
+        assert isinstance(safe_tx, (SafeTxV1, SafeTxV2))
+        confirmations = confirmations or list(self.client.get_confirmations(safe_tx_id))
+        if (confirmations_needed := self.confirmations_required - len(confirmations)) <= 0:
+            raise ApeSafeError("Transaction has enough confirmations.")
+
+        elif not (
+            available_signers := [
+                acc
+                for acc in self.local_signers
+                if acc.address not in set(c.owner for c in confirmations)
+            ]
+        ):
+            raise ApeSafeError("No local signers available to sign.")
 
-    def select_signer(self, for_: str = "submitter") -> AccountAPI:
-        return select_account(prompt_message=f"Select a {for_}", key=self.local_signers)
+        new_signatures = {}
+        for acc in available_signers:
+            if signature := acc.sign_message(safe_tx):
+                new_signatures[acc.address] = signature
+                confirmations_needed -= 1
+                if confirmations_needed == 0:
```

Comment:
In case it is ever 0 required confirmations for some reason (I know that probably isn't possible but is just safer anyway)

```suggestion
                if confirmations_needed <= 0:
```